tfx/components/example_gen/utils.py

```diff
@@ -209,51 +213,96 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
   regex_pattern = regex_pattern.replace(')', '\\)')
   return regex_pattern
 
-
-def _retrieve_latest_span(uri: Text,
-                          split: example_gen_pb2.Input.Split) -> Text:
-  """Retrieves the most recently updated span matching a given split pattern."""
-  split_pattern = os.path.join(uri, split.pattern)
-  if split_pattern.count(SPAN_SPEC) != 1:
-    raise ValueError('Only one {SPAN} is allowed in %s' % split_pattern)
-
-  split_glob_pattern = split_pattern.replace(SPAN_SPEC, '*')
-  logging.info('Glob pattern for split %s: %s', split.name, split_glob_pattern)
-  split_regex_pattern = _glob_to_regex(split_pattern).replace(SPAN_SPEC, '(.*)')
-  logging.info('Regex pattern for split %s: %s', split.name,
-               split_regex_pattern)
+def _property_search(split: example_gen_pb2.Input.Split,
+                     split_glob_pattern: Text, split_regex_pattern: Text,
+                     property_name: Text) -> Text:
+  """Search for most recently updated property with glob and regex patterns."""
   if re.compile(split_regex_pattern).groups != 1:
     raise ValueError('Regex should have only one group')
 
   files = tf.io.gfile.glob(split_glob_pattern)
-  latest_span = None
+  latest_property = None
   for file_path in files:
     result = re.search(split_regex_pattern, file_path)
     if result is None:
       raise ValueError('Glob pattern does not match regex pattern')
     try:
-      span = int(result.group(1))
+      prop = int(result.group(1))
     except ValueError:
-      raise ValueError('Cannot not find span number from %s based on %s' %
-                       (file_path, split_regex_pattern))
-    if latest_span is None or span >= int(latest_span):
+      raise ValueError('Cannot not find %s number from %s based on %s' %
+                       (property_name, file_path, split_regex_pattern))
+    if latest_property is None or prop >= int(latest_property):
       # Uses str instead of int because of zero padding digits.
-      latest_span = result.group(1)
+      latest_property = result.group(1)
 
-  if latest_span is None:
+  if latest_property is None:
     raise ValueError('Cannot not find matching for split %s based on %s' %
                      (split.name, split.pattern))
-  return latest_span
+  return latest_property
+
+
+def _retrieve_latest_span_version(uri: Text,
+                                  split: example_gen_pb2.Input.Split
+                                  ) -> Tuple[Text, Text]:
+  """Retrieves the most recent span and version for a given split pattern"""
```

Comment:
nit, update the docstring given the cases is more complex than before.

Reply:
added a doc string to `_retrieve_latest_span_version()`.